packages/salesforcedx-vscode-soql/src/client/codeCompletion.ts

```diff
@@ -142,34 +136,49 @@ const expandFunctions: {
       return [];
     }
 
-    const sobjectFields = objMetadata.fields.reduce((fieldItems, field) => {
-      fieldItems.push(
-        newCompletionItem(
-          field.name,
-          field.name,
-          CompletionItemKind.Field,
-          field.type
-        )
-      );
-      if (field.relationshipName) {
+    const sobjectFields = objMetadata.fields
```

Comment:
Since the checked in `filter` are simple and isolated you could move it to being the first part of `reduce` and make this a bit more efficient.

Reply:
Thanks. Done: 395b8f16134c0b4a1da338558c0fb0f6374c41ea